File: frontend/src/types/voice.ts
export interface Voice {
  id: number;
  name: string;
  elevenlabs_voice_id: string;
  description: string | null;
  is_cloned: boolean;
  created_at: string;
}

export interface CreateVoiceInput {
  name: string;
  elevenlabs_voice_id: string;
  description?: string | null;
  is_cloned?: boolean;
}

export interface CloneVoiceInput {
  name: string;
  description?: string | null;
  samples: File[];
}

export interface UpdateVoiceInput {
  name: string;
  description?: string | null;
}

File: backend/app/models/voice.py
from datetime import UTC, datetime

from sqlmodel import Field, SQLModel


class Voice(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    name: str
    elevenlabs_voice_id: str
    description: str | None = None
    is_cloned: bool = False
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
